backend/de.metas.contracts/src/main/java/de/metas/contracts/commission/commissioninstance/services/hierarchy/CommissionHierarchyFactory.java

```diff
@@ -36,20 +36,36 @@
  * #L%
  */
 
-/** Crates a hierarchy of BPartners by recursively following {@code C_BPartner.C_BPartner_SalesRep_ID} references. */
+/**
+ * Crates a hierarchy of BPartners by recursively following {@code C_BPartner.C_BPartner_SalesRep_ID} references.
+ */
 @Service
 public class CommissionHierarchyFactory
 {
-	// very crude but simple implementation; expand and make more efficient as needed
-	public Hierarchy createFor(@NonNull final BPartnerId bPartnerId)
+	@NonNull
+	public Hierarchy createForCustomer(@NonNull final BPartnerId bPartnerId, @NonNull final BPartnerId salesRepId)
 	{
+		if (bPartnerId.equals(salesRepId))
```

Comment:
not sure, this is just a hunch: why don't we recurse with the sales rep-id but directly return?

Reply:
good point, I'll change the logic to support also this scenario. Did you got a change to look over the docu, I've addressed a question there.